grpc/federation/context.go

```diff
@@ -22,6 +23,35 @@ func Logger(ctx context.Context) *slog.Logger {
 	return value.(*slog.Logger)
 }
 
+// SetLogger set logger instance for current context.
+// This is intended to be called from a custom resolver and is currently propagated to the current context and its children.
+func SetLogger(ctx context.Context, logger *slog.Logger) {
+	value := ctx.Value(customResolverValueKey{})
+	if value == nil {
+		return
+	}
+	value.(*CustomResolverValue).Logger = logger
+}
+
+// CustomResolverValue type for storing references to be modified when calling a custom resolver.
+type CustomResolverValue struct {
+	Logger *slog.Logger
+}
+
+func WithCustomResolverValue(ctx context.Context) context.Context {
```

Comment:
May I ask why we need this struct? If all we want to pass to custom resolvers is the logger, users can extract the logger directly from the context using `grpcfed.Logger`. Are we going to pass other values as well in the future?

Reply:
That's right! I've made it a struct so it can be easily extended in the future.
Also, user cannot directly rewrite the logger instance within the custom resolver.